plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -290,7 +298,10 @@ public RootRef getRootRef() {
       purRepositoryServiceRegistry.registerService( ILockService.class, new UnifiedRepositoryLockService( pur ) );
       purRepositoryServiceRegistry
         .registerService( IAclService.class, new UnifiedRepositoryConnectionAclService( pur ) );
-      metaStore = new PurRepositoryMetaStore( this );
+      metaStore = metastoreLocator.getExplicitMetastore( "VfsMetastoreProvider" );
```

Comment:
Use constant `MetastoreLocator.VFS_PROVIDER_KEY` ?

Reply:
The reason I didn't use this constant is because it requires a new dependency that I am trying to avoid since it caused problems before. I could make it constant for this class though that's a good idea